src/ig/IGExporter.ts

```diff
@@ -1479,6 +1589,69 @@ export class IGExporter {
       }
     }
   }
+
+  // Add an extension for the R5 property IG.definition.page.source[x] if provided in configuration
+  addNameUrlAndPageSourceExtensionForR4(
+    page: ImplementationGuideDefinitionPage,
+    configPages: ImplementationGuideDefinitionPage[]
+  ): void {
+    const sourceExtensionUrl =
+      'http://hl7.org/fhir/5.0/StructureDefinition/extension-ImplementationGuide.definition.page.source';
+    const nameExtensionUrl =
+      'http://hl7.org/fhir/5.0/StructureDefinition/extension-ImplementationGuide.definition.page.name';
+    const configPage = configPages?.find(
+      p =>
+        (p.nameUrl || p.name).substring(0, (p.nameUrl || p.name).lastIndexOf('.')) ===
+        page.nameUrl.substring(0, page.nameUrl.lastIndexOf('.'))
```

Comment:
I think this might be easier to read if done using `replace` rather than `substring`.  What do you think?
```suggestion
        (p.nameUrl || p.name).replace(/\.[^.]+$/, '') === page.nameUrl.replace(/\.[^.]+$/, '')
```
(Also, I didn't test this, so if you like it you should probably test it locally just to be sure it works).